emperor/support_files/js/color-view-controller.js

```diff
@@ -193,71 +204,154 @@ define([
+   *
+   * Retrieve a scaled color set with scale bar.
+   *
+   * @param {values} list of objects to generate a color for, usually a
+   * category in a given metadata column.
+   * @param {map} string, name of the discrete color map to use.
+   *
+   * This function will generate a list of coloring values depending on the
+   * coloring scheme that the system is currently using (discrete or
+   * continuous).
+   *
+   * Defaults to use viridis colormap if there's no map passed in.
+   *
+   */
+  ColorViewController.getScaledColors = function(values, map, nanColor) {
+    var map = map || 'viridis';
+    var nanColor = nanColor || '#64655d';
+    map = chroma.brewer[map];
+
+    // Get list of only numeric values, error if none
+    var numericValues = values.filter(function(v) { return !isNaN(v)});
+    if (numericValues.length < 2) {
+      alert('less than 2 numeric values found, can not create scale for category!');
+      throw new Error('non-numeric category');
+    }
+    numericValues = _.map(numericValues, Number);
+    min = _.min(numericValues);
+    max = _.max(numericValues);
+    interpolator = chroma.scale(map).domain([min, max]);
+    var colors = {};
+    for (var i = 0; i < values.length; i++) {
+      var val = Number(values[i]);
+      if (!isNaN(val)) {
+        colors[values[i]] = interpolator(val).hex();
       } else {
-        colors[values[index]] = interpolator(counter / numColors).hex();
-        counter = counter + 1;
+        //Gray out non-numeric values
+        colors[values[i]] = nanColor;
       }
     }
-
-    // if scaled, generate the scale and add to div.
-    if (scaled) {
-      var min = _.min(numericValues);
-      var max = _.max(numericValues);
-      var mid = (min + max) / 2;
-      step = (max - min) / 100;
-      var stopColors = [];
-      for (var s = min; s <= max; s += step) {
-        stopColors.push(interpolator(s).hex());
-      }
-
-      var gradientSVG = '<defs><linearGradient id="Gradient" x1="0" x2="1" y1="1" y2="1">';
-      for (pos in stopColors) {
-        gradientSVG += '<stop offset="' + pos + '%" stop-color="' + stopColors[pos] + '"/>';
-      }
-      gradientSVG += '</defs><rect id="gradientRect" width="100%" height="20" fill="url(#Gradient)"/>';
-      gradientSVG += '<text x="0%" y="38" font-family="sans-serif" font-size="12px">' + min + '</text>';
-      gradientSVG += '<text x="50%" y="38" font-family="sans-serif" font-size="12px" text-anchor="middle">' + mid + '</text>';
-      gradientSVG += '<text x="100%" y="38" font-family="sans-serif" font-size="12px" text-anchor="end">' + max + '</text>';
+    //build the SVG showing the gradient of colors for values
+    var mid = (min + max) / 2;
+    var step = (max - min) / 100;
+    var stopColors = [];
+    for (var s = min; s <= max; s += step) {
+      stopColors.push(interpolator(s).hex());
     }
-
+    var gradientSVG = '<defs><linearGradient id="Gradient" x1="0" x2="1" y1="1" y2="1">';
+    for (pos in stopColors) {
```

Comment:
Just noticed this, can you change this iteration to use the indices instead of the `in` keyword?